File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.js
var addSpaces = function(s, spaces) {
    let ans = [];
    let k = spaces.length;
    let j = 0;

    for (let i = 0; i < s.length; i++) {
        if (j < k && i === spaces[j]) {
            ans.push(" ");
            j++;
        }
        ans.push(s[i]);
    }

    return ans.join("");
};

File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder ans = new StringBuilder();
        int k = spaces.length;
        int j = 0;

        for (int i = 0; i < s.length(); i++) {
            if (j < k && i == spaces[j]) {
                ans.append(" ");
                j++;
            }
            ans.append(s.charAt(i));
        }

        return ans.toString();
    }
}